be1-go/internal/popserver/database/mock_repository.go

```diff
@@ -0,0 +1,966 @@
+// Code generated by mockery v2.42.1. DO NOT EDIT.
+
+package database
+
+import (
+	message "popstellar/message/query/method/message"
+
+	mock "github.com/stretchr/testify/mock"
+	kyber "go.dedis.ch/kyber/v3"
+
+	types "popstellar/internal/popserver/types"
+)
+
+// MockRepository is an autogenerated mock type for the Repository type
+type MockRepository struct {
+	mock.Mock
+}
+
+// CheckPrevCreateOrCloseID provides a mock function with given fields: channel, nextID
+func (_m *MockRepository) CheckPrevCreateOrCloseID(channel string, nextID string) (bool, error) {
+	ret := _m.Called(channel, nextID)
+
+	if len(ret) == 0 {
+		panic("no return value specified for CheckPrevCreateOrCloseID")
+	}
+
+	var r0 bool
+	var r1 error
+	if rf, ok := ret.Get(0).(func(string, string) (bool, error)); ok {
+		return rf(channel, nextID)
+	}
+	if rf, ok := ret.Get(0).(func(string, string) bool); ok {
+		r0 = rf(channel, nextID)
+	} else {
+		r0 = ret.Get(0).(bool)
+	}
+
+	if rf, ok := ret.Get(1).(func(string, string) error); ok {
+		r1 = rf(channel, nextID)
+	} else {
+		r1 = ret.Error(1)
+	}
+
+	return r0, r1
+}
+
+// CheckPrevOpenOrReopenID provides a mock function with given fields: channel, nextID
+func (_m *MockRepository) CheckPrevOpenOrReopenID(channel string, nextID string) (bool, error) {
+	ret := _m.Called(channel, nextID)
+
+	if len(ret) == 0 {
+		panic("no return value specified for CheckPrevOpenOrReopenID")
+	}
+
+	var r0 bool
+	var r1 error
+	if rf, ok := ret.Get(0).(func(string, string) (bool, error)); ok {
+		return rf(channel, nextID)
+	}
+	if rf, ok := ret.Get(0).(func(string, string) bool); ok {
+		r0 = rf(channel, nextID)
+	} else {
+		r0 = ret.Get(0).(bool)
+	}
+
+	if rf, ok := ret.Get(1).(func(string, string) error); ok {
+		r1 = rf(channel, nextID)
+	} else {
+		r1 = ret.Error(1)
+	}
+
+	return r0, r1
+}
+
+// GetAllMessagesFromChannel provides a mock function with given fields: channelID
+func (_m *MockRepository) GetAllMessagesFromChannel(channelID string) ([]message.Message, error) {
+	ret := _m.Called(channelID)
+
+	if len(ret) == 0 {
+		panic("no return value specified for GetAllMessagesFromChannel")
+	}
+
+	var r0 []message.Message
+	var r1 error
+	if rf, ok := ret.Get(0).(func(string) ([]message.Message, error)); ok {
+		return rf(channelID)
+	}
+	if rf, ok := ret.Get(0).(func(string) []message.Message); ok {
+		r0 = rf(channelID)
+	} else {
+		if ret.Get(0) != nil {
+			r0 = ret.Get(0).([]message.Message)
+		}
+	}
+
+	if rf, ok := ret.Get(1).(func(string) error); ok {
+		r1 = rf(channelID)
+	} else {
+		r1 = ret.Error(1)
+	}
+
+	return r0, r1
+}
+
+// GetChannelType provides a mock function with given fields: channel
+func (_m *MockRepository) GetChannelType(channel string) (string, error) {
```

Comment:
Try to rename the variables to clearer ones and add more comments to explain what it actually does inside because it's not very clear.